lib/id_partner.rb

```diff
@@ -0,0 +1,278 @@
+# frozen_string_literal: true
+
+require_relative "ruby_oidc_client/version"
+require "json/jwt"
+require "net/http"
+require "uri"
+require "base64"
+
+module RubyOidcClient
+  class IDPartner
+    SUPPORTED_AUTH_METHODS = [
+      "client_secret_basic",
+      "tls_client_auth",
+      "private_key_jwt" # For backward compatibility
+    ].freeze
+    SIGNING_ALG = "PS256"
+    ENCRYPTION_ALG = "RSA-OAEP"
+    ENCRYPTION_ENC = "A256CBC-HS512"
+
+    attr_reader :config, :endpoints, :provider_keys
+
+    def initialize(config)
+      raise ArgumentError, "Config missing." unless config
+
+      default_config = {
+        account_selector_service_url: "https://auth-api.idpartner.com/oidc-proxy",
+        token_endpoint_auth_method: "client_secret_basic",
+        jwks: nil,
+        client_secret: nil
+      }
+
+      @config = default_config.merge(config)
+
+      unless SUPPORTED_AUTH_METHODS.include?(@config[:token_endpoint_auth_method])
+        raise ArgumentError,
+              "Unsupported token_endpoint_auth_method '#{config[:token_endpoint_auth_method]}'. It must be one of (#{SUPPORTED_AUTH_METHODS.join(", ")})"
+      end
+
+      client_secret_config = @config[:token_endpoint_auth_method] == "client_secret_basic" ? { client_secret: @config[:client_secret] } : {}
+
+      jwks_config = if @config[:jwks]
+                      {
+                        authorization_encrypted_response_alg: ENCRYPTION_ALG,
+                        authorization_encrypted_response_enc: ENCRYPTION_ENC,
+                        id_token_encrypted_response_alg: ENCRYPTION_ALG,
+                        id_token_encrypted_response_enc: ENCRYPTION_ENC,
+                        request_object_signing_alg: SIGNING_ALG
+                      }
+                    else
+                      {}
+                    end
+
+      @config = @config.merge({
+                                client_id: @config[:client_id],
+                                token_endpoint_auth_method: @config[:token_endpoint_auth_method],
+                                redirect_uri: @config[:redirect_uri],
+                                authorization_signed_response_alg: SIGNING_ALG,
+                                id_token_signed_response_alg: SIGNING_ALG
+                              }).merge(client_secret_config).merge(jwks_config)
+    end
+
+    def generate_proofs
+      {
+        state: SecureRandom.urlsafe_base64(64),
+        nonce: SecureRandom.urlsafe_base64(64),
+        code_verifier: SecureRandom.urlsafe_base64(64)
+      }
```

Comment:
check node to use the same

Reply:
We are doing the same, but instead of 32 i am using 64:
```
irb(main):050:0> SecureRandom.urlsafe_base64(32).length
=> 43

> Buffer.isEncoding('base64url')
true
> Buffer.from(randomBytes(32), 'utf8').toString('base64url').length
43
```
https://github.com/panva/node-openid-client/blob/main/lib/helpers/generators.js#L5